lib/ansible/module_utils/service.py

```diff
@@ -207,17 +207,19 @@ def daemonize(module, cmd):
         p = subprocess.Popen(run_cmd, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE, preexec_fn=lambda: os.close(pipe[1]))
         fds = [p.stdout, p.stderr]
 
-        # loop reading output till its done
+        # loop reading output till it is done
         output = {p.stdout: b(""), p.stderr: b("")}
         while fds:
             rfd, wfd, efd = select.select(fds, [], fds, 1)
-            if (rfd + wfd + efd) or p.poll():
+            if (rfd + wfd + efd) or p.poll() is None:
                 for out in list(fds):
                     if out in rfd:
                         data = os.read(out.fileno(), chunk)
                         if not data:
                             fds.remove(out)
-                    output[out] += b(data)
+                        output[out] += data
+            else:
+                break
```

Comment:
I think you misunderstand why this loop is here, the while waits either until the descriptors are populated OR the program fails (poll() returns true aka rc > 0).

Reply:
looking back i think i made a mistake when merging the previous PR

Reply:
@bcoca this is a backport of #81064, I explained and verified these changes there.

Reply:
@bcoca both issues are very easy to reproduce: `service` module with `daemonize: true` simply doesn't work.

Reply:
reread that, but i'm rethinking at least part of it as the loop changes meaning this way, output assignment to data is still wrong imho, should be but in the else and then we can always ensure it is binary again.